### PR TITLE
`2515` Update typing and simplify DAO loading logic

### LINKED ISSUE #2515: `[TypeFix]` | Update `DAOInfo` typing and get implementation

### PR DESCRIPTION
**Pull request by Da-Colon**
Closes #2515 

During the screenshare earlier, QA'ing the work on stores, Noticed that there was an extra `address` property while loading DAOInfo, along with some other weird typing. Continuing from there noticed the similar logic and disconnect between loading the initial DAO info and the Organization page. 

This improves I believe the whole setup, deletes a bunch of code and looks like loading is quite smoother(er?)

Check out Myosin's Organization page to stress test the loading.

`/hierarchy?dao=base:0x0c14aA2d34C0E13Bbe441963e4de2EeAef4abeD2`

### CODE REVIEW

File: src/hooks/DAO/loaders/useLoadDAONode.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+import { DAO, DAOQueryDocument } from '../../../../.graphclient';
 import { useSafeAPI } from '../../../providers/App/hooks/useSafeAPI';
 import { useNetworkConfig } from '../../../providers/NetworkConfig/NetworkConfigProvider';
-import { DaoInfo, Node, WithError } from '../../../types';
-import { mapChildNodes } from '../../../utils/hierarchy';
+import { DaoInfo } from '../../../types';
 import { useGetSafeName } from '../../utils/useGetSafeName';
 import { useFractalModules } from './useFractalModules';
 
+type DAOInfoWithoutLoaders = Omit<DaoInfo, 'isModulesLoaded' | 'isHierarchyLoaded'>;
```
Comment (DarksightKellar) [inline src/hooks/DAO/loaders/useLoadDAONode.ts:11]: Just putting this out there for the overhaul work we're doing, and whenever we get around to cleaning this up -- these (`isXyxLoaded`) are actually state data and should be stored separately in state, not on the DaoInfo object itself.

File: src/utils/hierarchy.ts
Comment (DarksightKellar) [inline src/utils/hierarchy.ts:1]: Love it

**Review by DarksightKellar — APPROVED** [review]

File: src/hooks/DAO/loaders/useLoadDAONode.ts (hunk 1 above)
Comment (Da-Colon) [reply src/hooks/DAO/loaders/useLoadDAONode.ts:11]: Hopefully we just remove them completely when store swap is done :)

File: src/hooks/DAO/loaders/useLoadDAONode.ts [hunk 2]
```diff
@@ -23,59 +23,63 @@ export const useLoadDAONode = () => {
...
-    };
-    return currentNode;
-  }, []);
+  const getDAOWithHierarchyData = useCallback(
+    async (dao: DAO) => {
+      const childData: DAOInfoWithoutLoaders[] = [];
+      for await (const child of dao.hierarchy) {
+        const safe = await safeAPI.getSafeData(getAddress(child.address));
+        childData.push({
+          fractalModules: [],
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/hooks/DAO/loaders/useLoadDAONode.ts:32]: Yeah. until we look into the require docs and https://www.notion.so/hulking-cosmos-3e8/Q4-2024-QA-Requirements-13e2ee9c5bc280a88458e7a935d17488?pvs=4#13f2ee9c5bc2801bbacad5d7e7fd659e

`Add voting schema “ERC-20”, “ERC-721”, “Multisig” to the listed subdaos`

Not for this PR but...

[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/hooks/DAO/loaders/useLoadDAONode.ts:32]: We need the modules -> then do all the checks for what type of DAO and then what type of Governance if Azorious for each DAO.

**Pull request merged**